odo/backends/url.py

```diff
@@ -133,7 +133,9 @@ def append_urlX_to_X(target, source, **kwargs):
         with open(target.path, 'wb') as fp:
             for chunk in iter(curry(r.read, chunk_size), ''):
                 fp.write(chunk)
-                return target
+                if not chunk:
```

Comment:
We definitely hit this branch, without it, it's an infinite loop :)

The issue is that `http.client.HTTPResponse.read` returns `b''` forever when it's finished, so we have to explicitly test for this case and break.

We could check for `len(chunk) < chunk_size`, but this code here seems cleaner to me.

I'll add a comment explaining `HTTPResponse.read`'s behavior here.